jwst/coron/median_replace_img.py

```diff
@@ -0,0 +1,77 @@
+"""Replace bad pixels with the median of the surrounding pixel and median fill
+   smooth the psf images.
+ """
+import logging
+import numpy as np
+
+from jwst.datamodels import dqflags
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+def median_fill_value(input_array, input_dq_array, bsize, xc, yc):
+    """
+        Arguments:
+        ----------
+        input_array : ndarray
+            Input array to filter.
+        input_dq_array : ndarray
+            Input data quality array
+        bsize : scalar
+            box size of the data to extract
+        xc: scalar
+           x position of the data extraction
+        xc: scalar
+           y position of the data extraction
+        """
+    # set the half box size
+    hbox = int(bsize/2)
+
+    # Extract the region of interest for the data
+    try:
+        data_array = input_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+        dq_array = input_dq_array[xc - hbox:xc + hbox, yc - hbox: yc + hbox]
+    except IndexError:
+        # If the box is outside the data return 0
+        log.warning('Box for median filter is outside the data.')
+        return 0.
```

Comment:
Hmmm, so this is not going to allow for replacement of any bad pixels that're within half the box width of the image borders, instead replacing them with a value of zero. Given that later on in the alignment a weight/mask is applied that essentially gives zero weight to pixels near the image perimeter, hopefully the fact that they haven't had a real replacement done to their values will be OK. I guess we'll see what happens in testing.

Reply:
That is what I was counting on here. In general I'd use the median of the outer part of the image, or perhaps the outermost values of a radial profile.